core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/metrics/datasizes/DetermineDatalakeDataSizesActions.java

```diff
@@ -0,0 +1,68 @@
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackFailureEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.datalakemetrics.datasizes.GetDatalakeDataSizesRequest;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.datalakemetrics.datasizes.DetermineDatalakeDataSizesSuccessEvent;
+import com.sequenceiq.cloudbreak.service.StackUpdater;
+
+@Configuration
+public class DetermineDatalakeDataSizesActions {
+    @Inject
+    private StackUpdater stackUpdater;
+
+    @Bean(name = "DETERMINE_DATALAKE_DATA_SIZES_IN_PROGRESS_STATE")
+    public Action<?, ?> determineDatalakeDataSizesAction() {
+        return new AbstractDetermineDatalakeDataSizesAction<>(StackEvent.class) {
+            @Override
+            protected void doExecute(DetermineDatalakeDataSizesContext context, StackEvent payload, Map<Object, Object> variables) {
+                stackUpdater.updateStackStatus(
+                        context.getStackId(), DetailedStackStatus.DETERMINE_DATALAKE_DATA_SIZES_IN_PROGRESS, "Determination of datalake data sizes in progress"
+                );
+                sendEvent(context);
+            }
+
+            @Override
+            protected Selectable createRequest(DetermineDatalakeDataSizesContext context) {
+                return new GetDatalakeDataSizesRequest(context.getStackId());
+            }
+        };
+    }
+
+    @Bean(name = "DETERMINE_DATALAKE_DATA_SIZES_SUCCESS_STATE")
+    public Action<?, ?> determineDatalakeDataSizesSuccessAction() {
+        return new AbstractDetermineDatalakeDataSizesAction<>(DetermineDatalakeDataSizesSuccessEvent.class) {
+            @Override
+            protected void doExecute(DetermineDatalakeDataSizesContext context, DetermineDatalakeDataSizesSuccessEvent payload, Map<Object, Object> variables) {
+                stackUpdater.updateStackStatus(
+                        context.getStackId(), DetailedStackStatus.DETERMINE_DATALAKE_DATA_SIZES_FINISHED, payload.getDataSizesResult()
+                );
+                sendEvent(context, DetermineDatalakeDataSizesEvent.DETERMINE_DATALAKE_DATA_SIZES_SUCCESS_HANDLED_EVENT.event(), payload);
+            }
+        };
+    }
+
+    @Bean(name = "DETERMINE_DATALAKE_DATA_SIZES_FAILURE_STATE")
+    public Action<?, ?> determineDatalakeDataSizesFailureAction() {
+        return new AbstractDetermineDatalakeDataSizesAction<>(StackFailureEvent.class) {
+            @Override
+            protected void doExecute(DetermineDatalakeDataSizesContext context, StackFailureEvent payload, Map<Object, Object> variables) {
+                getFlow(context.getFlowId()).setFlowFailed(payload.getException());
+                stackUpdater.updateStackStatus(
+                        context.getStackId(), DetailedStackStatus.DETERMINE_DATALAKE_DATA_SIZES_FAILED, "Determining the datalake data sizes failed"
```

Comment:
This will be marked as a failure state in the UI because of the keyword "FAILED". Can you confirm it?
we don't want that. I what I stated is true, we can update the status to DETERMINE_DATALAKE_DATA_SIZES_FINISHED.

Reply:
The status is mapped to the available status so wouldn't this be fine?
In `DetailedStackStatus` there's `DETERMINE_DATALAKE_DATA_SIZES_FAILED(Status.AVAILABLE)`